pydicom/tests/test_handler_util.py

```diff
@@ -752,3 +774,623 @@ def test_length_ybr_422(self, shape, bits, length):
+        assert [0] * 128 == out[:128]
+        assert [246, 248, 250, 252, 254] == out[-5:]
+
+    def test_palettes_fall(self):
+        """Test expanding the FALL palette."""
+        ds = dcmread(get_palette_files('fall.dcm')[0])
+
+        bs = ds.SegmentedRedPaletteColorLookupTableData
+        fmt = '<{}B'.format(len(bs))
+        data = unpack(fmt, bs)
+        out = _expand_segmented_lut(data, fmt)
+        assert [255] * 256 == out
+
+        bs = ds.SegmentedGreenPaletteColorLookupTableData
+        fmt = '<{}B'.format(len(bs))
+        data = unpack(fmt, bs)
+        out = _expand_segmented_lut(data, fmt)
+        assert list(range(255, -1, -1)) == out
+
+        bs = ds.SegmentedBluePaletteColorLookupTableData
+        fmt = '<{}B'.format(len(bs))
+        data = unpack(fmt, bs)
+        out = _expand_segmented_lut(data, fmt)
+        assert [0] * 256 == out
+
+    def test_palettes_winter(self):
+        """Test expanding the WINTER palette."""
+        ds = dcmread(get_palette_files('winter.dcm')[0])
+
+        bs = ds.SegmentedRedPaletteColorLookupTableData
+        fmt = '<{}B'.format(len(bs))
+        data = unpack(fmt, bs)
+        out = _expand_segmented_lut(data, fmt)
+        assert [0] * 128 == out[:128]
+        assert [123, 124, 125, 126, 127] == out[-5:]
+
+        bs = ds.SegmentedGreenPaletteColorLookupTableData
+        fmt = '<{}B'.format(len(bs))
+        data = unpack(fmt, bs)
+        out = _expand_segmented_lut(data, fmt)
+        assert list(range(0, 256)) == out
+
+        bs = ds.SegmentedBluePaletteColorLookupTableData
+        fmt = '<{}B'.format(len(bs))
+        data = unpack(fmt, bs)
+        out = _expand_segmented_lut(data, fmt)
+        assert [255, 255, 254, 254, 253] == out[:5]
+        assert [130, 129, 129, 128, 128] == out[-5:]
+
+    def test_first_linear_raises(self):
+        """Test having a linear segment first raises exception."""
+        data = (1, 5, 49152)
+        msg = (
+            r"Error expanding a segmented palette color lookup table: "
+            r"the first segment cannot be a linear segment"
+        )
+        with pytest.raises(ValueError, match=msg):
+            _expand_segmented_lut(data, 'H')
+
+    def test_first_indirect_raises(self):
+        """Test having a linear segment first raises exception."""
+        data = (2, 5, 2, 0)
+        msg = (
+            r"Error expanding a segmented palette color lookup table: "
+            r"the first segment cannot be an indirect segment"
+        )
+        with pytest.raises(ValueError, match=msg):
+            _expand_segmented_lut(data, 'H')
+
+    def test_unknown_opcode_raises(self):
+        """Test having an unknown opcode raises exception."""
+        data = (3, 5, 49152)
+        msg = (
+            r"Error expanding a segmented palette lookup table: "
+            r"unknown segment type '3'"
+        )
+        with pytest.raises(ValueError, match=msg):
+            _expand_segmented_lut(data, 'H')
```

Comment:
Note that I had only a cursory glance at the tests, only checking what is tested but not verifying the actual data - I trust that to be correct...